library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -61,25 +70,48 @@ public void setup(PortContext context)
     }
   };
 
-  @OutputPortFieldAnnotation(optional = true)
   public transient DefaultOutputPort<INPUT> err = new DefaultOutputPort<INPUT>();
```

Comment:
Instead of err<INPUT>, I feel error port with KeyValPair<STRING,STRING>  would be better as we could give the error reason along with that.